addons/intel/e825.go

```diff
@@ -19,195 +19,294 @@ import (
-			_, unitTest = (*nodeProfile).PtpSettings["unitTest"]
-			if unitTest {
-				MockPins()
-			}
 
-			if e825Opts.EnableDefaultConfig {
-				stdout, _ := exec.Command("/usr/bin/bash", "-c", EnableE825PTPConfig).Output()
-				glog.Infof(string(stdout))
-			}
-			if (*nodeProfile).PtpSettings == nil {
-				(*nodeProfile).PtpSettings = make(map[string]string)
-			}
+	if (*nodeProfile).PtpSettings == nil {
+		(*nodeProfile).PtpSettings = make(map[string]string)
+	}
 
-			// Prefer ZL3073x module for e825
-			zlClockID, zlErr := getClockIDByModule("zl3073x")
-			if zlErr != nil {
-				glog.Errorf("e825: failed to resolve ZL3073x DPLL clock ID via netlink: %v", zlErr)
-			}
-			for device, pins := range e825Opts.DevicePins {
-				dpllClockIDStr := fmt.Sprintf("%s[%s]", dpll.ClockIdStr, device)
-				if !unitTest {
-					if zlErr == nil {
-						(*nodeProfile).PtpSettings[dpllClockIDStr] = strconv.FormatUint(zlClockID, 10)
-					} else {
-						(*nodeProfile).PtpSettings[dpllClockIDStr] = strconv.FormatUint(getClockIDE825(device), 10)
-					}
-					for pin, value := range pins {
-						deviceDir := fmt.Sprintf("/sys/class/net/%s/device/ptp/", device)
-						phcs, pErr := os.ReadDir(deviceDir)
-						if pErr != nil {
-							glog.Error("e825 failed to read " + deviceDir + ": " + pErr.Error())
-							continue
-						}
-						for _, phc := range phcs {
-							pinPath := fmt.Sprintf("/sys/class/net/%s/device/ptp/%s/pins/%s", device, phc.Name(), pin)
-							glog.Infof("echo %s > %s", value, pinPath)
-							err = os.WriteFile(pinPath, []byte(value), 0o666)
-							if err != nil {
-								glog.Error("e825 failed to write " + value + " to " + pinPath + ": " + err.Error())
-							}
-						}
-					}
-				}
-			}
+	if tbcConfigured(nodeProfile) {
+		// For T-BC, default to GNSS=disabled, but allow manual override in the user-speciifed config
+		e825Opts.Gnss.Disabled = true
+	}
 
-			for k, v := range e825Opts.DpllSettings {
-				if _, ok := (*nodeProfile).PtpSettings[k]; !ok {
-					(*nodeProfile).PtpSettings[k] = strconv.FormatUint(v, 10)
-				}
-			}
-			for iface, properties := range e825Opts.PhaseOffsetPins {
-				ifaceFound := false
-				for dev := range e825Opts.DevicePins {
-					if strings.Compare(iface, dev) == 0 {
-						ifaceFound = true
-						break
-					}
-				}
-				if !ifaceFound {
-					glog.Errorf("e825 phase offset pin filter initialization failed: interface %s not found among  %v",
-						iface, reflect.ValueOf(e825Opts.DevicePins).MapKeys())
-					break
-				}
-				for pinProperty, value := range properties {
-					var clockIDUsed uint64
-					if zlErr == nil {
-						clockIDUsed = zlClockID
-					} else {
-						clockIDUsed = getClockIDE825(iface)
-					}
-					key := strings.Join([]string{iface, "phaseOffsetFilter", strconv.FormatUint(clockIDUsed, 10), pinProperty}, ".")
-					(*nodeProfile).PtpSettings[key] = value
-				}
+	// Parse user-specified config
+	opts, ok := nodeProfile.Plugins[pluginNameE825]
```

Comment:
This map lookup is better than iterating through the map looking for the key, and because of this we can unindent everything by 2 levels